src/main/java/org/sagebionetworks/web/client/widget/header/Header.java

```diff
@@ -95,4 +93,20 @@ public void onLoginClick() {
 	public void onRegisterClick() {
 		globalApplicationState.getPlaceChanger().goTo(new RegisterAccount(ClientProperties.DEFAULT_PLACE_TOKEN));	
 	}
+
+	@Override
+	public void onFavoriteClick() {
+		List<EntityHeader> headers = globalApplicationState.getFavorites();
+		view.clearFavorite();
+		if (headers == null || headers.size() == 0) {
+			view.setEmptyFavorite();
+		} else {
+			view.addFavorite(headers);
+		}
+	}
+
+	@Override
+	public void applyFavClick(EntityHeader header) {
+		globalApplicationState.getPlaceChanger().goTo(new Synapse(header.getId()));
```

Comment:
try setting the anchor list item href to DisplayUtils.getSynapseHistoryToken() for each entity, rather than js handling.